pilot/test/integration/driver_test.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"testing"
+	"os"
+	"istio.io/istio/pkg/log"
+)
+
+// This is a golang test wrapper around our integration test driver.
+func TestEverything(t *testing.T) {
+	_ = log.Configure(log.NewOptions())
+
+	params = newInfra()
+
+	// Only run the tests if the user has defined the KUBECONFIG environment variable.
+	params.KubeConfig = os.Getenv("KUBECONFIG")
+	if len(params.KubeConfig) == 0 {
+		log.Info("Env variable KUBECONFIG not set. Skipping integration tests")
+		return
+	}
+
+	// By default, all the paths are set from the top-level Istio directory. Need to adjust these for unit tests
+	// to be relative to the current working directory.
+	params.TestDataDir = "testdata/"
+	params.CertDir = "../../docker/certs/"
+
+	// Configure the docker HUB to use for images.
+	// TODO(nmittler): Use local HUB and build images locally.
+	params.Hub = "gcr.io/istio-testing"
+	params.Tag = "54a3ecc8ae3e72c919d63436d0d12a7c5f3600c8"
+
+	// Run all the tests.
+	if err := runTests(*params); err != nil {
```

Comment:
Can you pass t as parameter, and have the runTest method do t.Run() ? 
If I read this correctly, it'll show up as a single big golang test - which is not much better than the original. Fine to do this in a separate PR, of course - but if something is planned please add TODOs so reviewers know...

Reply:
Absolutely ... good idea.  I've restructured things so that this test is no longer dependent on driver.go ... instead, it restructures the `runTests` method to use `testing.T`.